src/esp/gfx_batch/Renderer.h

```diff
@@ -695,6 +695,11 @@ class Renderer {
   Corrade::Containers::StridedArrayView1D<Magnum::Float> lightRanges(
       Magnum::UnsignedInt sceneId);
 
+  /**
+   * @brief Draw all scenes in the default framebuffer.
+   */
+  virtual void draw();
```

Comment:
Hmm, I don't like the `virtual` here, plus the implicit reliance on the default framebuffer. The low-level API should be as explicit as possible, and this function goes against that.

Why can't the `BatchReplayRenderer` just call `draw(GL::defaultFramebuffer)` on its own? Then it wouldn't need a new `draw()` overload at all.